apps/backend/app.py

```diff
@@ -185,12 +201,37 @@ def upload_experiment_results(expId, trialExtraFile, postProcess):
     uploadBlob = firebaseBucket.blob(f"results/result{expId}.csv")
     uploadBlob.upload_from_filename('results.csv')
 
+        # Upload to MongoDB
+
+    try:
+            mongoClient.admin.command('ping')
+    except ConnectionFailure:
+        print("Server not available (this is an error)")
+        
+    
+    print('Uploading to MongoDB')
+    experimentFile = open(f"results.csv") # there is probably a better way to do this
+    experimentData = experimentFile.read()
+    experimentFile.close()
+    experimentResultEntry = {"_id": expId,
+                        "resultContent": experimentData}
+    
+    resultId = mongoResultsCollection.insert_one(experimentResultEntry).inserted_id
+    print(f"inserted result csv into mongodb with id: {resultId}")
```

Comment:
Can this ever fail? possibly wrap in a try-catch and re-raise the exception with more details

Reply:
exception reraised